lib/SILOptimizer/Utils/CanonicalizeInstruction.cpp

```diff
@@ -548,6 +548,9 @@ static SILBasicBlock::iterator
 eliminateUnneededForwardingUnarySingleValueInst(SingleValueInstruction *inst,
                                                 CanonicalizeInstruction &pass) {
   auto next = std::next(inst->getIterator());
+  if (inst->getType().isMoveOnly()) {
```

Comment:
Do you want to use `isValueTypeWithDeinit` here or are you using `isMoveOnly` for now for the same reason given for using it in `forwardToInit`, i.e. "For simplicity and bug-finding, consider all non-Copyable types to have a deinit."?

Reply:
Indeed isValueTypeWithDeinit is better here. Fixed and broadly rewrote comments throughout.